infra/gravity/testsuite.go

```diff
@@ -170,26 +170,43 @@ func (s *testSuite) Schedule(fn TestFunc, cfg ProvisionerConfig, param interface
 }
 
 func (s *testSuite) getLogLink(testUID string) (string, error) {
-	longUrl := url.URL{
+	// TODO(dmitri): decide whether it would make sense to migrate to Firebase Dynamic Links as a replacement.
+	// In the past this method used Google URL shortener to create a short link. However, this service
+	// was discontinued: https://developers.googleblog.com/2018/03/transitioning-google-url-shortener.html
+	// For now, return full URLs.
+	longURL := encodeLogLink(testUID, s.uid, s.googleProjectID, time.Now())
+	return longURL, nil
+}
+
+// encodeLogLink formats test metadata into a clickable link to GCP logs.
+//
+// Logic split out from getLogLink to aid in unit testing.
+//
+// The resulting url will look like:
+//   https://console.cloud.google.com/logs/query;query=severity%3E%3DINFO%0Alabels.__uuid__%3D%22504d3d56-1abe-43cb-a802-3ebc96367d47%22%0Alabels.__suite__%3D%22d165f1b1-f40e-4e5f-8014-7bbb713d5357%22;timeRange=2020-09-22T22:33:00.000Z%2F2020-09-22T22:35:00.000Z?authuser=0&project=kubeadm-167321
+func encodeLogLink(tUID, sUID, project string, date time.Time) string {
+	// severity%3E%3DINFO%0Alabels.__uuid__%3D%22504d3d56-1abe-43cb-a802-3ebc96367d47%22%0Alabels.__suite__%3D%22d165f1b1-f40e-4e5f-8014-7bbb713d5357%22
+	query := fmt.Sprintf(`severity>=INFO
+labels.__uuid__="%s"
```

Comment:
Instead of `"%s"` here and below you can just use `%q` which is a quoted `%s`.

Reply:
Updated in 10ddc58.